src/github/controller.py

```diff
@@ -42,6 +45,63 @@ def _add_asana_task_to_pull_request(pull_request: PullRequest, task_id: str):
     pull_request.set_body(new_body)
 
 
+def upsert_subtasks(pull_request: PullRequest, task_id: str) -> None:
+    """
+    Create subtasks for the reviewers.
+
+    This is synchronized with dynamodb so we only create a subtask once for each reviewer.
+
+    We go through the list of requested reviewers and check if they have a subtask created already.
+    If not, we create the subtask.
+    """
+    for reviewer in pull_request.requested_reviewers():
+        subtask_id = dynamodb_client.get_asana_id_from_two_github_node_ids(
+            pull_request.id(), reviewer.id()
+        )
+        if subtask_id is None:
+            created_subtask_id = asana_controller.create_review_subtask(
+                pull_request, task_id, reviewer.login()
+            )
+            if created_subtask_id is None:
+                # TODO: Handle this case
+                logger.error(
+                    f"No subtask id returned from create subtask {pull_request.id()}: {task_id}"
+                )
+                continue
+
+            logger.info(
+                f"Subtask created for pull request {pull_request.id()}: {task_id}: {created_subtask_id}"
+            )
+            dynamodb_client.insert_two_github_node_to_asana_id_mapping(
+                pull_request.id(), reviewer.id(), created_subtask_id
+            )
+
+
+def update_subtasks(pull_request: PullRequest, task_id: str) -> None:
```

Comment:
I think it'd be more efficient to do this and `upsert_subtasks` in one loop rather than two separate functions. Mostly because it would cut the calls to dynamodb in half and updating wouldn't be necessary for newly created subtasks